test/e2e-common.sh

```diff
@@ -67,6 +67,36 @@ function test_setup() {
   wait_until_pods_running istio-system || return 1
   echo ">> Waiting for Serving components to be running..."
   wait_until_pods_running knative-serving || return 1
+
+  # Wait for static IP to be through
+  wait_for_static_ip istio-system istio-ingressgateway
+}
+
+function wait_for_static_ip() {
+  local ns=$1
+  local svc=$2
+  local sleep_seconds=20
+  local attempts=100
+
+  STATIC_IP=""
+  for try in $(seq 1 $attempts); do
+    STATIC_IP=$(kubectl get svc $svc -n $ns -ojsonpath={.status.loadBalancer.ingress[0].ip})
+    STATUS_CODE=""
+    if [[ $STATIC_IP != "" ]]; then
+      echo ">> Probing $STATIC_IP"
+      STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" http://$STATIC_IP)
+      if [[ $STATUS_CODE != "" && $STATUS_CODE != "000" ]]; then
```

Comment:
do we consider 5xx also success?

Reply:
Yes, this is only testing TCP connection to confirm that the underlying network setup is done.  Any non-zero HTTP status code is good indicator here.